### PR TITLE
Resolve most compiler warnings

### PR DESCRIPTION
**Pull request by serebit**
## Description
Resolve most compiler warnings in Budgie source, while only disabling one warning. This generally consists of the following:

- Adding try/catch when it's necessary
- Replacing `variable as Type` with `(Type) variable` in places where the safe cast was being ignored anyway
- Removing unused functions and variables
- Replacing deprecated functions with their non-deprecated equivalents (thanks GNOME devs)
- Changing semantics so that the Vala transpiler stops complaining
- Adding `throws DBusError` to DBus functions even if they don't throw DBusError

I look forward to the inevitable change requests.

### Submitter Checklist

- [x] Squashed commits with `git rebase -i` (if needed)
- [x] Built budgie-desktop and verified that the patch worked (if needed)

### CODE REVIEW

**Review by JoshStrobl — CHANGES_REQUESTED** [review]

File: src/applets/user-indicator/DBusInterfaces.vala [hunk 1]
```diff
@@ -9,32 +9,32 @@
...
-public interface LogindInterface : GLib.Object {
-	public abstract void suspend(bool interactive) throws IOError;
-	public abstract void hibernate(bool interactive) throws IOError;
+[DBus (name = "org.freedesktop.login1.Manager")]
+public interface LogindInterface : Object {
+	public abstract void suspend(bool interactive) throws DBusError, IOError;
+	public abstract void hibernate(bool interactive) throws DBusError, IOError;
 }
 
 [DBus (name="org.gnome.SessionManager")]
```
[the remark this answers is not in the document]
Comment (serebit) [reply src/applets/user-indicator/DBusInterfaces.vala:35]: Those were accidental, I must have missed them in my review pass.

File: src/raven/headerwidget.vala [hunk 2]
```diff
@@ -14,7 +14,7 @@ namespace Budgie {
 	* Simple expander button for the header widget
 	*/
 	public class HeaderExpander : Gtk.Button {
-		private Gtk.Image? image;
+		private new Gtk.Image? image;
```
[the remark this answers is not in the document]
Comment (serebit) [reply src/raven/headerwidget.vala:17]:
```
../src/raven/headerwidget.vala:17.3-17.26: warning: Budgie.HeaderExpander.image hides inherited field `Gtk.Button.image'. Use the `new' keyword if hiding was intentional
```

File: src/daemon/menus.vala [hunk 3]
```diff
@@ -144,7 +144,7 @@ namespace Budgie {
 		/**
 		* Show a window menu for the given window ID
 		*/
-		public void ShowWindowMenu(uint32 xid, uint button, uint32 timestamp) {
+		public void ShowWindowMenu(uint32 xid, uint button, uint32 timestamp) throws Error {
```
[the remark this answers is not in the document]
Comment (serebit) [reply src/daemon/menus.vala]: Corrected to `throws DBusError, IOError` in latest push.

File: src/daemon/osd.vala [hunk 4]
```diff
@@ -226,7 +226,7 @@ namespace Budgie {
 		* level: Progress-level to display in the OSD (double or int32 depending on gsd release)
 		* monitor: int32 The monitor to display the OSD on (currently ignored)
 		*/
-		public void ShowOSD(HashTable<string,Variant> params) {
+		public void ShowOSD(HashTable<string,Variant> params) throws Error {
```
[the remark this answers is not in the document]
Comment (serebit) [reply src/daemon/osd.vala:229]:
```
../src/daemon/menus.vala:147.3-147.28: warning: DBus methods are recommended to throw at least `GLib.Error' or `GLib.DBusError, GLib.IOError'
```
Corrected to `throws DBusError, IOError` in latest push.

[the remark this answers is not in the document]
Comment (JoshStrobl) [reply src/daemon/osd.vala:229]: Maybe it's GitHub being derp but still shows as `throw Error` here.

**Review by JoshStrobl — CHANGES_REQUESTED** [review]

File: src/daemon/osd.vala (hunk 4 above)
[the remark this answers is not in the document]
Comment (serebit) [reply src/daemon/osd.vala]: It wasn't, I missed this function in particular. Just pushed a fix.

File: src/plugin/popover.c [hunk 5]
```diff
@@ -724,9 +722,6 @@ static gboolean budgie_popover_draw(GtkWidget* widget, cairo_t* cr) {
 	gtk_widget_get_allocation(widget, &alloc);
 	body_alloc = alloc;
 
-	/* Set up the offset */
-	gdouble gap_start = 0, gap_end = 0;
```
[the remark this answers is not in the document]
Comment (serebit) [reply src/plugin/popover.c:728]: image
image
image
image
image

**Comment by JoshStrobl** [comment]
LGTM, thank you for all the fixes!

**Pull request merged**